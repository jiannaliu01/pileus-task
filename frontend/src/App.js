import React, { Component } from "react";
import NavBar from "./components/NavBar";
import { Route, Redirect } from "react-router-dom";
import LoginView from "./components/LoginView";
import RegisterView from "./components/RegisterView";
import Statistics from "./components/Statistics";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      redirect: null
    };
  }
  render() {
    return (
      <div>
        <NavBar />
        <Route exact path="/" component={LoginView} />
        <Route exact path="/register" component={RegisterView} />
        <Route exact path="/statistics" component={Statistics} />
      </div>
    );
  }
}

export default App;
