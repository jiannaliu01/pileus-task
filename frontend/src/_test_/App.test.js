import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import App from "../App";

afterEach(cleanup);

it("matches snapshot", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
