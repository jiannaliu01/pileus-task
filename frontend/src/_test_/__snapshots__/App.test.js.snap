// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-dark bg-primary fixed-top"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Pileus Assignment
      </a>
    </nav>
    <div
      class="container"
    >
      <div
        class="row mb-5"
      >
        <div
          class="col-lg-12 text-center"
        >
          <h1
            class="mt-5"
          >
            Login
          </h1>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-12"
        >
          <form
            action="#"
          >
            <div
              class="form-group"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                class="form-control "
                name="email"
                placeholder="Enter email"
                type="email"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                class="form-control "
                name="password"
                placeholder="Enter password"
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-primary btn-block"
              type="submit"
            >
              Login
            </button>
            <button
              class="btn btn-primary btn-block"
              type="submit"
            >
              Have Not Logged In? Register!
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
